packages/siwx/src/ui/w3m-data-capture-view/index.ts

```diff
@@ -0,0 +1,296 @@
+      if (event.key === 'Enter') {
+        this.onSubmit()
+      }
+    }
+
+    const changeHandler = (event: CustomEvent<string>) => {
+      this.email = event.detail
+    }
+
+    return html`
+      <wui-flex flexDirection="column">
+        <wui-email-input
+          .value=${this.email}
+          .disabled=${this.loading}
+          @inputChange=${changeHandler}
+          @keydown=${keydownHandler}
+        ></wui-email-input>
+
+        <w3m-email-suffixes-widget
+          .email=${this.email}
+          @change=${changeHandler}
+        ></w3m-email-suffixes-widget>
+      </wui-flex>
+    `
+  }
+
+  private recentEmailsWidget() {
+    if (this.recentEmails.length === 0 || this.loading) {
+      return null
+    }
+
+    const recentEmailSelectHandler = (event: CustomEvent<string>) => {
+      this.email = event.detail
+      this.onSubmit()
+    }
+
+    return html`
+      <w3m-recent-emails-widget
+        .emails=${this.recentEmails}
+        @select=${recentEmailSelectHandler}
+      ></w3m-recent-emails-widget>
+    `
+  }
+
+  private footerActions() {
+    return html`
+      <wui-flex flexDirection="row" fullWidth gap="s">
+        ${this.isRequired
+          ? null
+          : html`<wui-button
+              size="lg"
+              variant="neutral"
+              fullWidth
+              .disabled=${this.loading}
+              @click=${this.onSkip.bind(this)}
+              >Skip this step</wui-button
+            >`}
+
+        <wui-button
+          size="lg"
+          variant="main"
+          type="submit"
+          fullWidth
+          .disabled=${!this.email || !this.isValidEmail(this.email)}
+          .loading=${this.loading}
+          @click=${this.onSubmit.bind(this)}
+        >
+          Continue
+        </wui-button>
+      </wui-flex>
+    `
+  }
+
+  private async onSubmit() {
+    const account = ChainController.getActiveCaipAddress()
+
+    if (!account) {
+      throw new Error('Account is not connected.')
+    }
+
+    if (!this.isValidEmail(this.email)) {
+      SnackController.showError('Please provide a valid email.')
+
+      return
+    }
+
+    try {
+      this.loading = true
+      const otp = await this.siwx.requestEmailOtp({
+        email: this.email,
+        account
+      })
+
+      this.pushRecentEmail(this.email)
+
+      if (otp.uuid === null) {
+        RouterController.replace('SIWXSignMessage')
+      } else {
+        RouterController.replace('DataCaptureOtpConfirm', { email: this.email })
+      }
+
+      await new Promise(resolve => {
+        // Wait for view transition
+        setTimeout(resolve, 300)
+      })
+    } catch (error) {
+      SnackController.showError('Failed to send email OTP')
+    } finally {
+      this.loading = false
+    }
+  }
+
+  private onSkip() {
+    RouterController.replace('SIWXSignMessage')
+  }
+
+  private getRecentEmails() {
+    const recentEmails = SafeLocalStorage.getItem(SafeLocalStorageKeys.RECENT_EMAILS)
```

Comment:
is this only used here? we are trying not to read from storage in the views.

Reply:
Proper way would probably be to do it during intitialization but not sure you have a state where you persist siwx things